crates/wallet/src/lib.rs

```diff
@@ -251,82 +243,59 @@ where
         // we acquire the permit here so that all following operations are performed exclusively
         let _permit = self.inner.permit.lock().await;
 
-        // set nonce
-        let next_nonce = LoadState::next_available_nonce(
-            &self.inner.eth_api,
-            NetworkWallet::<Ethereum>::default_signer_address(&self.inner.wallet),
-        )
-        .await
-        .map_err(|err| {
-            self.inner.metrics.invalid_send_transaction_calls.increment(1);
-            err.into()
-        })?;
-        request.nonce = Some(next_nonce);
-
         // set chain id
         request.chain_id = Some(self.chain_id());
 
         // set gas limit
         // note: we also set the `from` field here to correctly estimate for contracts that use e.g.
         // `tx.origin`
-        request.from = Some(NetworkWallet::<Ethereum>::default_signer_address(&self.inner.wallet));
+
+        // todo: what to do here
+        // request.from =
+        // Some(NetworkWallet::<Ethereum>::default_signer_address(&self.inner.wallet));
         let (estimate, base_fee) = tokio::join!(
-            EthCall::estimate_gas_at(&self.inner.eth_api, request.clone(), BlockId::latest(), None),
-            LoadFee::eip1559_fees(&self.inner.eth_api, None, None)
+            self.inner.provider.estimate_gas(&request),
+            self.inner.provider.estimate_eip1559_fees(None)
         );
-        let estimate = estimate.map_err(|err| {
+        let estimate = estimate.map_err(|_err| {
             self.inner.metrics.invalid_send_transaction_calls.increment(1);
-            err.into()
+            // err.into()
+            OdysseyWalletError::InvalidTransactionRequest
```

Comment:
need to fix the error handling, it's a bit cumbersome to work with `RpcResult` using `thiserror`